blocks/init/src/Blocks/components/button/components/button-options.js

```diff
@@ -81,7 +82,7 @@ export const ButtonOptions = (attributes) => {
 								</Fragment>
 							}
 							value={buttonColor}
-							colors={getOptionColors(options.colors)}
+							colors={getOptionColors(getOptions(manifest, componentName, 'color', options))}
```

Comment:
I think it would be more readable if all these `getOptions` calls were extracted out of the `return` code 🤷🏼‍♂️ 

This applies to the changes on lines 99 and 113 as well.

Reply:
sure but I would like to have all the component props at one place so you can reuse them easily

Reply:
Not sure I understand. What would be different if the `getOptions(manifest, componentName, 'color', options)` was set as `const colorOptions` and then passed here?

Reply:
if you have a big component you need to scroll up and down to see the configuration and what it does. Also all component options are scoped under the condition so this way this helper will not fire if it is not in use. 

Your way you need to put a variable on the top, scroll up and down to read what component does and you need to also put a if condition on the variable on the top so you duplicate your work.

Reply:
> Also all component options are scoped under the condition so this way this helper will not fire if it is not in use.

If you have just one place where you use it, sure, but if you have it in multiple places chances are that you'll use it. It's imported anyhow, so I don't think it will be huge a performance loss.

> Your way you need to put a variable on the top, scroll up and down to read what component does and you need to also put a if condition on the variable on the top so you duplicate your work.

Wrapper has this, tons of spacing options are defined at the top, and used below in the components. I really don't have a big problem with this.

I'll always choose duplication over the loss of readability tbh 🤷🏼‍♂️  

I'm fine to leave it in, it just looks super hard to understand what is happening here, especially since it's wrapped in `getOptionColors` helper.